src/components/organisms/OakPagination/OakPagination.tsx

```diff
@@ -0,0 +1,218 @@
+    color: ${props.selected ? "white" : "black"};
+  `}
+`;
+
+const StyledEllipsis = styled(InternalButton)`
+  &:disabled {
+    cursor: pointer;
+  }
+`;
+
+const OakPageNumber = ({
+  currentPage,
+  pageIndex,
+  onClick,
+  href,
+  pageName,
+}: OakPageNumberProps) => {
+  const isActive = currentPage === pageIndex;
+  return (
+    <StyledNumberButton
+      data-testid="page-number-component"
+      aria-label={`${pageName} page ${pageIndex}`}
+      aria-current={isActive ? "page" : false}
+      $font={"heading-7"}
+      onClick={onClick}
+      selected={isActive}
+      href={href}
+    >
+      {pageIndex}
+    </StyledNumberButton>
+  );
+};
+
+const OakEllipsis = () => {
+  return (
+    <StyledEllipsis
+      $color="text-primary"
+      $font={"heading-7"}
+      $background="white"
+      aria-label="Hidden page numbers"
+      disabled
+    >
+      ...
+    </StyledEllipsis>
+  );
+};
+
+export const OakPagination = ({
+  currentPage,
+  totalPages,
+  nextHref,
+  prevHref,
+  paginationHref,
+  pageName,
+}: OakPaginationProps) => {
+  const [activePage, setActivePage] = useState(currentPage);
+
+  const pageNumbers = generatePageNumbers(activePage, totalPages);
+
+  const isFirstPage = activePage <= 1;
+  const isLastPage = activePage >= totalPages;
+
+  const handleNumberClick = (num: number) => {
+    setActivePage(num);
+  };
+
+  const handleChevronClick = (direction: "backwards" | "forwards") => {
+    setActivePage((currNum) => currNum + (direction === "backwards" ? -1 : 1));
+  };
+
+  if (currentPage === 1 && totalPages < 2) {
+    return null;
+  }
+  return (
+    <nav aria-label="pagination" data-testid="pagination">
+      <OakFlex
+        $alignItems={"center"}
+        $justifyContent={"center"}
+        $gap={["space-between-ssx", "space-between-s", "space-between-s"]}
+      >
+        <StyledChevronButton
+          element={isFirstPage ? "button" : "a"}
+          data-testid="backwards-button"
+          onClick={() => {
+            handleChevronClick("backwards");
+          }}
+          href={prevHref}
+          aria-disabled={isFirstPage}
+          disabled={activePage <= 1}
+          aria-label={isFirstPage ? "No previous pages" : "Go to previous page"}
+        >
+          <StyledIcon disabled={isFirstPage} iconName="chevron-left" />
+        </StyledChevronButton>
+        <OakUL $reset $display={"flex"}>
+          {pageNumbers.map((pageIndex, i) => {
+            if (typeof pageIndex === "number") {
+              return (
+                <OakLI
+                  key={`${pageIndex} ${i}`}
+                  $mh={[
+                    "space-between-sssx",
+                    "space-between-ssx",
+                    "space-between-ssx",
+                  ]}
+                >
+                  <OakPageNumber
+                    pageName={pageName}
+                    key={pageIndex}
+                    pageIndex={pageIndex + 1}
```

Comment:
it's a bit inconsistent how pageindex and pageNumber are named throughout, this should be pageNumber if it's the index +1? similarly we're mapping over indices but they're called numbers

Reply:
✅